ui/components/app/assets/defi-list/defi-list.tsx

```diff
@@ -0,0 +1,259 @@
+          address: '0x',
+          title: protocol.protocolDetails.name,
+          symbol: protocol.protocolDetails.name,
+          tokenFiatAmount: protocol.aggregatedMarketValue,
+          image: protocol.protocolDetails.iconUrl,
+          primary: protocol.aggregatedMarketValue.toString(),
+          secondary: protocol.aggregatedMarketValue,
+          decimals: 10,
+          chainId: chainId as '0x' & string,
+          iconGroup,
+        };
+      }),
+    );
+
+    const filteredAssets = filterAssets(defiProtocolCells, [
+      {
+        key: 'chainId',
+        opts: networkFilter,
+        filterCallback: 'inclusive',
+      },
+    ]);
+
+    // sort filtered tokens based on the tokenSortConfig in state
+    return sortAssets(filteredAssets, tokenSortConfig);
+  }, [allDefiPositions, networkFilter, selectedAccount, tokenSortConfig]);
+
+  const handleTokenClick = (token: DeFiProtocolPosition) => () => {
+    onClick(token.chainId, token.protocolId);
+
+    trackEvent({
+      category: MetaMetricsEventCategory.DeFi,
+      event: MetaMetricsEventName.DefiDetailsOpened,
+      properties: {
+        location: 'Home',
+        chain_id: token.chainId,
+        protocol_id: token.protocolId,
+      },
+    });
+  };
+
+  if (sortedFilteredDefi === undefined) {
+    return (
+      <Box
+        display={Display.Flex}
+        flexDirection={FlexDirection.Column}
+        alignItems={AlignItems.center}
+        justifyContent={JustifyContent.center}
+      >
+        <PulseLoader />
+      </Box>
+    );
+  }
+
+  if (sortedFilteredDefi === null) {
+    return (
+      <ErrorMessage
+        title={t('defiTabErrorTitle')}
+        text={t('defiTabErrorContent')}
+      />
+    );
+  }
+
+  return (
+    <>
+      {sortedFilteredDefi && sortedFilteredDefi?.length > 0 ? (
+        sortedFilteredDefi.map((token: DeFiProtocolPosition) => {
+          return (
+            <TokenCell
+              key={`${token.chainId}-${token.protocolId}`}
+              token={token}
+              privacyMode={privacyMode}
+              onClick={handleTokenClick(token)}
+              primaryDisplayOverride={() => (
+                <AvatarGroup
+                  avatarType={AvatarType.TOKEN}
+                  limit={4}
+                  members={token.iconGroup}
+                  data-testid="defi-list-avatar-group"
+                />
+              )}
+              fixCurrencyToUSD={true}
+            />
+          );
+        })
+      ) : (
+        <TokenCell
+          key={`empty-defi-list`}
+          token={{
+            address: '0x',
+            title: 'Start earning',
+            symbol: 'Start earning',
+            tokenFiatAmount: 0,
+            image: `images/logo/metamask-fox.svg`,
+            primary: '0',
+            secondary: 0,
+            decimals: 10,
+            chainId: '0x1',
+            isStakeable: true,
+          }}
```

Comment:
Maybe we can save this object into a variable for clarity on what it represents.

Maybe `defiListFallbackToken` or something along those lines.